File: lunar_api/report/use_case/delete_report.py
# Copyright © 2026 Idiap Research Institute <contact@idiap.ch>
# SPDX-FileContributor: Danilo Gusicuma <danilo@lunarbase.ai>
#
# SPDX-License-Identifier: GPL-3.0-only

from uuid import UUID

from lunar_api.report.repository.report_repository import ReportRepository


class DeleteReportUseCase:
    def __init__(self, report_repository: ReportRepository):
        self._report_repository = report_repository

    def execute(self, report_id: UUID, user_id: str | None = None) -> bool:
        """Delete a report by ID.

        Args:
            report_id: The report UUID

        Returns:
            True if deleted successfully, False otherwise
        """
        return self._report_repository.delete_report(report_id=report_id, user_id=user_id)

File: lunar-chat-interface/src/actions/delete-report.ts
// SPDX-FileCopyrightText: 2026 Copyright © 2026 Idiap Research Institute <contact@idiap.ch>
// SPDX-FileContributor: Danilo Gusicuma <danilo@lunarbase.ai>
//
// SPDX-License-Identifier: GPL-3.0-only

"use server";

import { revalidatePath } from "next/cache";
import { serverApiUrl } from "@/configuration";
import { fetchWithServerAuth } from "@/lib/server-auth";

const basePath = process.env.NEXT_PUBLIC_BASE_PATH ?? "";

export async function deleteReportAction(reportId: string) {
  if (!serverApiUrl) {
    throw new Error("Server API URL is not configured");
  }

  const response = await fetchWithServerAuth(
    `${serverApiUrl}/report/${reportId}`,
    {
      method: "DELETE",
      headers: {
        "Content-Type": "application/json",
      },
    },
  );

  if (!response.ok) {
    const error = await response
      .json()
      .catch(() => ({ message: "Failed to delete report" }));
    throw new Error(error.message || "Failed to delete report");
  }

  revalidatePath(`${basePath}/app/report`);
  return await response.json();
}
